go/vt/vtgate/engine/merge.go

```diff
@@ -0,0 +1,72 @@
+package engine
+
+import (
+	"sync"
+
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/vtgate/queryinfo"
+)
+
+// Merge specifies the parameters for a merge primitive
+type Merge struct {
+	Left, Right Primitive `json:",omitempty"`
+}
+
+// Execute performs an execute
+func (merge *Merge) Execute(vcursor VCursor, queryConstruct *queryinfo.QueryConstruct, joinvars map[string]interface{}, wantfields bool) (*sqltypes.Result, error) {
+	var lres, rres *sqltypes.Result
+	var lerr, rerr error
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(vc *VCursor, qc *queryinfo.QueryConstruct, jv map[string]interface{}, wf bool) {
```

Comment:
I'm not sure what's more idiomatic, but looking at an example of channels it look like using a channel could be cleaner with only minimal more code.  This and the function below are identical except for Left vs Right. Using channels to return the values you could break this out into a regular function and call it with Left and Right. You can also pass in *wg as an arg to call Done().